Sources/UI/Views/TextPollCell.swift

```diff
@@ -0,0 +1,332 @@
+            self.resultFillBarArea.layoutIfNeeded()
+        })
+        
+        let percentageTextFont = self.option.text.font.bumpedForPercentageIndicator
+
+        // expand the percentage view to accomodate all possible percentage values as we animate through them, to avoid any possible wobble in the layout.
+        self.resultPercentageWidthConstraint?.constant = percentageTextFont.attributedText(forPlainText: "100%", color: self.option.text.color)?.boundingRect(with: .init(width: 1_000, height: 1_000), options: [], context: nil).width ?? CGFloat(0)
+        
+        self.percentageAnimationTimer?.invalidate()
+        let startTime = Date()
+        self.percentageAnimationTimer = Timer.scheduledTimer(withTimeInterval: 0.01, repeats: true, block: { [weak self] timer in
+            // TODO: calculate a "start position" from the current value on the constraint, in order for repeat calls to `revealResultsState` to properly animate through the percentages between the current value rather than just 0.
+            let elapsed = Double(startTime.timeIntervalSinceNow) * -1
+            let elapsedProportion = elapsed / RESULT_FILL_BAR_FILL_TIME
+            if elapsedProportion > 1.0 {
+                self?.resultPercentage.text = String(format: "%.0f%%", optionResults.fraction * 100)
+                timer.invalidate()
+            } else {
+                self?.resultPercentage.text = String(format: "%.0f%%", Double(optionResults.fraction * 100) * elapsedProportion)
+            }
+        })
+    }
+    
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("Usage in XIB not supported.")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        // we defer configuring background image to here so that the layout has been calculated, and thus frame is available.
+        self.backgroundView.configureAsBackgroundImage(background: option.background)
+    }
+}
+
+// MARK: Cell View
+
+class TextPollCell: BlockCell {
+    private let containerView = UIView()
+    
+    private var optionViews = [TextPollOptionView]()
+    
+    override var content: UIView? {
+        return containerView
+    }
+    
+    private var questionView: PollQuestionView?
+    
+    private var temporaryTapDemoTimer: Timer?
+    
+    override func configure(with block: Block) {
+        super.configure(with: block)
+     
+        containerView.translatesAutoresizingMaskIntoConstraints = false
+        questionView?.removeFromSuperview()
+        self.optionViews.forEach { $0.removeFromSuperview() }
+        
+        guard let textPollBlock = block as? TextPollBlock else {
+            return
+        }
+    
+        questionView = PollQuestionView(questionText: textPollBlock.textPoll.question)
+        containerView.addSubview(questionView!)
+        questionView?.topAnchor.constraint(equalTo: containerView.topAnchor).isActive = true
```

Comment:
When creating and activating a number of constraints at the same time, consider using the static method: `NSLayoutConstraint.activate(_:)`.